crates/re_build_search_index/src/ingest/rust.rs

```diff
@@ -0,0 +1,512 @@
+        let reader = BufReader::new(file);
+        let krate: Crate = serde_json::from_reader(reader).unwrap();
+        crates.push(krate);
+    }
+
+    let (tx, rx) = mpsc::channel();
+    let version = ctx.rerun_pkg().version.clone();
+
+    ctx.finish_progress_bar(progress);
+
+    crates
+        .into_iter()
+        .map(|krate| {
+            (
+                ctx.progress_bar(format!("rustdoc ({})", krate.name())),
+                krate,
+            )
+        })
+        .collect::<Vec<_>>()
+        .into_par_iter()
+        .for_each(|(progress, krate)| {
+            let mut visitor = Visitor::new(progress, &version, &tx, &krate);
+            visitor.visit_root();
+            visitor.progress.finish_and_clear();
+        });
+
+    drop(tx);
+    for data in rx {
+        ctx.push(data);
+    }
+
+    Ok(())
+}
+
+struct Visitor<'a> {
+    progress: ProgressBar,
+    visited: HashSet<ItemId>,
+    documents: &'a mpsc::Sender<DocumentData>,
+    module_path: Vec<String>,
+    krate: &'a Crate,
+    base_url: String,
+}
+
+impl<'a> Visitor<'a> {
+    fn new(
+        progress: ProgressBar,
+        version: &Version,
+        documents: &'a mpsc::Sender<DocumentData>,
+        krate: &'a Crate,
+    ) -> Self {
+        let crate_name = krate.name();
+
+        Self {
+            progress,
+            visited: HashSet::new(),
+            documents,
+            krate,
+            module_path: vec![crate_name],
+            base_url: base_url(version, krate),
+        }
+    }
+
+    fn push(&mut self, pub_in_priv: bool, id: &ItemId, kind: ItemKind) {
+        let path = self.resolve_path(pub_in_priv, id);
+        self.push_with_path(id, kind, &path);
+    }
+
+    fn push_with_path(&mut self, id: &ItemId, kind: ItemKind, path: &[String]) {
+        // don't push the same document twice
+        if self.visited.contains(id) {
+            return;
+        }
+        self.visited.insert(id.clone());
+
+        let mut module_path = &path[..path.len() - 1];
+        let name = path.last().unwrap();
+        let item_path = match &kind {
+            ItemKind::Module => {
+                format!("{name}/index.html")
+            }
+            ItemKind::Struct
+            | ItemKind::Enum
+            | ItemKind::Trait
+            | ItemKind::Function
+            | ItemKind::Type
+            | ItemKind::Constant
+            | ItemKind::Macro => {
+                format!("{kind}.{name}.html")
+            }
+            ItemKind::Assoc(parent, _) => {
+                let parent_name = module_path.last().unwrap();
+                module_path = &module_path[..module_path.len() - 1];
+                format!("{parent}.{parent_name}.html#{kind}.{name}")
+            }
+        };
+
+        let _ = self.documents.send(document(
+            path.join("::"),
+            format!("{}/{}/{}", self.base_url, module_path.join("/"), item_path),
+            self.krate.index[id].docs.clone().unwrap_or_default(),
+        ));
```

Comment:
nit/style: we prefer `.ok()` when explicitly ignoring a `Result`: https://github.com/rerun-io/rerun/blob/main/CODE_STYLE.md#style


```suggestion
        self.documents.send(document(
            path.join("::"),
            format!("{}/{}/{}", self.base_url, module_path.join("/"), item_path),
            self.krate.index[id].docs.clone().unwrap_or_default(),
        )).ok();
```

Reply:
I will keep forgetting this because there's no lint for it, sorry in advance 😅